packages/expo-modules-core/android/src/main/java/expo/modules/interfaces/taskManager/TaskServiceProviderHelper.kt

```diff
@@ -0,0 +1,36 @@
+package expo.modules.interfaces.taskManager
+
+import android.content.Context
+import expo.modules.core.ModulePriorities
+import expo.modules.core.interfaces.Package
+
+class TaskServiceProviderHelper {
+  companion object {
+    /*
+      Uses reflection to look through the current list of packages and attempts to find
+      one that provides a TaskServiceInterface implementation.
+     */
+    fun getTaskServiceImpl(context: Context): TaskServiceInterface? {
+      // Use reflection to get the packages list from ExpoModulesPackageList without
+      // creating the reactInstanceManager:
+      val expoModules: Class<*>? = try {
+        Class.forName("expo.modules.ExpoModulesPackageList")
+      } catch (e: ClassNotFoundException) {
+        // Handle the exception, e.g., log it or fallback to a default behavior
+        return null
+      }
+      val getPackageList = expoModules?.getMethod("getPackageList") ?: return null
+
+      // Invoke and get the list of packages
+      val result = getPackageList.invoke(null) as? List<*> ?: return null
+      val packages = result.filterIsInstance<Package>()
```

Comment:
is this filter and sort needed?

Reply:
Not necessarily, but it is the way it is resolved when the app is loaded so I just wanted to make sure we didn't introduce any side effects by applying another sorting order.